packages/analyticsdx-template-lint/src/linter.ts

```diff
@@ -796,11 +796,11 @@ export abstract class TemplateLinter<
     const [autoInstallDef, autoInstallDefNode] = findJsonPrimitiveAttributeValue(tree, 'autoInstallDefinition');
     if (autoInstallDefNode && typeof autoInstallDef === 'string') {
       const [templateType, templateTypeNode] = findJsonPrimitiveAttributeValue(tree, 'templateType');
-      // only app and embeddedapp templates can specify an autoInstallDefinition
-      if (templateTypeNode && templateType !== 'app' && templateType !== 'embeddedapp') {
+      // only app, data and embeddedapp templates can specify an autoInstallDefinition
+      if (templateTypeNode && templateType !== 'app' && templateType !== 'embeddedapp' && templateType !== 'data') {
         this.addDiagnostic(
           doc,
-          "Only 'app' and 'embeddedapp' templates can use an 'autoInstallDefinition'",
+          "Only 'app', 'data' or 'embeddedapp' templates can use an 'autoInstallDefinition'",
```

Comment:
Since `data` is only valid in Winter '24 (246) and the warning is still correct for Summer '23 (244) and older, should this try to explain that? Something like `"Only 'app', 'embeddedapp', or 'data' (Winter '24) templates ..."`? Downside is that's kind of wordy, and maybe confusing, and we'll need to undo it later once Winter '24 is everywhere.
@mblumreich?